packages/model-viewer/src/three-components/GLTFInstance.ts

```diff
@@ -136,6 +136,13 @@ export class GLTFInstance implements GLTF {
         const materials =
             Array.isArray(mesh.material) ? mesh.material : [mesh.material];
         materials.forEach(material => {
+          // Explicitly dispose any textures assigned to this material
+          for (const propertyName in material) {
+            const propertyValue = (material as any)[propertyName];
+            if (propertyValue instanceof Texture) {
+              propertyValue.dispose();
```

Comment:
Good call! Not sure how we went this long without noticing our textures needed disposal. Did you verify this fixes the leak?

Reply:
Yes, the example I supplied in [#2139](https://github.com/google/model-viewer/issues/2139) doesn't create memory overflow when tested on my local build with proposed fix.